src/hotspot/share/classfile/stringTable.cpp

```diff
@@ -72,26 +73,31 @@ const size_t REHASH_LEN = 100;
 const double CLEAN_DEAD_HIGH_WATER_MARK = 0.5;
 
 #if INCLUDE_CDS_JAVA_HEAP
-inline oop read_string_from_compact_hashtable(address base_address, u4 offset) {
-  assert(ArchiveHeapLoader::are_archived_strings_available(), "sanity");
-  if (UseCompressedOops) {
-    assert(sizeof(narrowOop) == sizeof(offset), "must be");
-    narrowOop v = CompressedOops::narrow_oop_cast(offset);
-    return ArchiveHeapLoader::decode_from_archive(v);
+bool StringTable::_two_dimensional_shared_strings_array = false;
```

Comment:
Suggestion: `_is_two_dimensional`

Reply:
Renamed to `_two_dimensional_shared_strings_array`.